spec/services/hyrax/custom_queries/navigators/find_files_spec.rb

```diff
@@ -0,0 +1,25 @@
+RSpec.describe Hyrax::CustomQueries::Navigators::FindFiles do
+  let(:query_service) { Valkyrie::MetadataAdapter.find(:test_adapter).query_service }
+
+  describe '#find_files' do
+    subject { query_service.custom_queries.find_files(resource: fileset) }
+    context 'when files exist' do
+      let!(:file_metadata1) { FactoryBot.create_using_test_adapter(:hyrax_file_metadata) }
+      let!(:file_metadata2) { FactoryBot.create_using_test_adapter(:hyrax_file_metadata) }
+      let!(:fileset) { FactoryBot.create_using_test_adapter(:hyrax_pcdm_file_set, files: [file_metadata1, file_metadata2]) }
+      it 'returns file metadata resource' do
+        expect(subject).to be_a Array
```

Comment:
is this part of the query contract? would any `Enumerable` be adequate?

Reply:
If the code is refactored, it could be any Enumerable.  The way it is written now, an Array is created and will be returned.  I would be inclined to create an issue that modifies all custom queries that return multiple results to return Enumerable rather than just change it here.

Reply:
tests shouldn't test things we expect to change under refactors. this makes the tests fragile, and introduces risk to refactoring (if i need to rewrite the tests to refactor, i run the risk of changing the expected behavior inadvertently).

I don't think this type check is appropriate.

Reply:
DONE - removed check for Array in all tests related to this PR